ui/push-health/App.jsx

```diff
@@ -19,26 +29,94 @@ function hasProps(search) {
   return params.get('repo') && params.get('revision');
 }
 
-const App = () => {
-  return (
-    <div>
-      <div>
+class App extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      user: { isLoggedIn: false },
+      notifications: [],
+    };
+  }
+
+  notify = (message, severity, options = {}) => {
+    const { notifications } = this.state;
+    const notification = {
+      ...options,
+      message,
+      severity: severity || 'darker-info',
+      created: Date.now(),
+    };
+    const newNotifications = [notification, ...notifications];
+
+    this.setState({
+      notifications: newNotifications,
+    });
+  };
+
+  clearNotification = (index = null) => {
+    const { notifications } = this.state;
+
+    if (index) {
+      this.setState(clearNotificationAtIndex(notifications, index));
+    } else {
+      this.setState(clearExpiredTransientNotifications(notifications));
+    }
+  };
+
+  render() {
+    const { user, notifications } = this.state;
+    const { path } = this.props.match;
+
+    return (
+      <ErrorBoundary
+        errorClasses={errorMessageClass}
+        message={genericErrorMessage}
+      >
+        <Navigation
+          user={user}
+          setUser={(user) => this.setState({ user })}
```

Comment:
This is totally fine.  But at some point, we should take advantage of the `useState` hook for things like this.